pkg/operator/target_config_reconciler.go

```diff
@@ -324,7 +364,271 @@ func (c TargetConfigReconciler) sync() error {
+				errorList = append(errorList, err)
+			} else {
+				klog.Infof("%s %s deleted successfully.", webhook.name, name)
+			}
+		}
+	}
+
+	if len(errorList) > 0 {
+		return utilerror.NewAggregate(errorList)
+	}
+	return nil
+}
+
+func (c *TargetConfigReconciler) cleanUpCustomResources(ctx context.Context) error {
+	var errorList []error
+	crdList, err := c.crdClient.CustomResourceDefinitions().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("Failed to list CRDs: %v", err)
+		return err
+	} else {
+		for _, crd := range crdList.Items {
+			if strings.Contains(crd.Name, "kueue") {
+				gvr := schema.GroupVersionResource{
+					Group:    crd.Spec.Group,
+					Version:  crd.Spec.Versions[0].Name,
+					Resource: crd.Spec.Names.Plural,
+				}
+
+				klog.Infof("Fetching instances of CR type: %s", crd.Name)
+
+				crList, err := c.dynamicClient.Resource(gvr).Namespace("").List(ctx, metav1.ListOptions{})
+				if err != nil {
+					klog.Errorf("Failed to list instances of %s: %v", crd.Name, err)
+					errorList = append(errorList, err)
+					continue
+				}
+
+				for _, cr := range crList.Items {
+					klog.Infof("Deleting Custom Resource: %s/%s", cr.GetNamespace(), cr.GetName())
+
+					err := retry.OnError(retry.DefaultBackoff, errors.IsTooManyRequests, func() error {
+						return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+							return c.dynamicClient.Resource(gvr).Namespace(cr.GetNamespace()).Delete(ctx, cr.GetName(), metav1.DeleteOptions{})
+						})
+					})
+					if err != nil {
+						klog.Errorf("Failed to delete CR %s/%s: %v", cr.GetNamespace(), cr.GetName(), err)
+						errorList = append(errorList, err)
+					} else {
+						klog.Infof("Successfully deleted CR: %s/%s", cr.GetNamespace(), cr.GetName())
+					}
+				}
+
+				err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+					latestCRD, err := c.crdClient.CustomResourceDefinitions().Get(ctx, crd.Name, metav1.GetOptions{})
+					if err != nil {
+						return err
+					}
+
+					if len(latestCRD.Finalizers) > 0 {
+						latestCRD.Finalizers = []string{}
+						_, err := c.crdClient.CustomResourceDefinitions().Update(ctx, latestCRD, metav1.UpdateOptions{})
+						return err
+					}
+					return nil
+				})
+				if err != nil {
+					klog.Errorf("Failed to remove finalizers for CRD %s: %v", crd.Name, err)
+					errorList = append(errorList, err)
+				}
+
+				err = retry.OnError(retry.DefaultBackoff, errors.IsTooManyRequests, func() error {
+					return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+						return c.crdClient.CustomResourceDefinitions().Delete(ctx, crd.Name, metav1.DeleteOptions{})
+					})
+				})
+
+				if err != nil {
+					klog.Errorf("Failed to delete CRD %s: %v", crd.Name, err)
+					errorList = append(errorList, err)
+				} else {
+					klog.Infof("Successfully deleted CRD: %s", crd.Name)
+				}
+			}
+		}
+	}
 
+	certManagerCRDs := []string{
+		"certificates.cert-manager.io",
```

Comment:
We should not delete cert-manager.

Reply:
Correct. I think we should delete the namespaced values for certificates and issuers upon deletion of the operator.

But cert manager CRDs should be left alone.